gramjs/client/uploads.ts

```diff
@@ -98,12 +98,10 @@ const DISCONNECT_SLEEP = 1000;
 async function getFileBuffer(
     file: File | CustomFile,
     fileSize: number,
-    maxBufferSize?: number
+    maxBufferSize: number
 ): Promise<CustomBuffer> {
-    const isBiggerThan2Gb = fileSize > 2 ** 31 - 1;
-    const usePath = maxBufferSize ? fileSize > maxBufferSize : false;
     const options: CustomBufferOptions = {};
-    if ((usePath || isBiggerThan2Gb) && file instanceof CustomFile) {
+    if (fileSize > maxBufferSize && file instanceof CustomFile) {
```

Comment:
no no that's not what I meant. 
just at the start of the function check if maxBufferSize is undefined and if it is you just do maxBufferSize = 2GB
the rest of the logic stays the same as you did before.